src/Link.tsx

```diff
@@ -10,23 +16,33 @@ type Props = LinkProps;
  * This will load the target JSResources on hover or focus, and will trigger
  * the loader, which will start requesting graphql data, on mouse down.
  */
-export default function Link({ onMouseEnter, onFocus, onMouseDown, ...props }: Props) {
+export default function Link({
+  onMouseEnter,
+  onFocus,
+  onMouseDown,
+  ...props
+}: Props) {
+  const fetchEntryPoint = useLinkEntryPointLoadHandler(props.to);
   const fetchResources = useLinkResourceLoadHandler(props.to);
   const fetchData = useLinkDataLoadHandler(props.to);
- 
+
+  useEffect(() => {
+    fetchEntryPoint();
```

Comment:
While I do like the idea of preloading the possible entry points early, I worry that doing it on link mount will be expensive on initial page load. e.g. if all my navigation is defined using links with entry points this will try to fetch all of the entrypoints as soon as the page mounts, which is already pretty contested with other network fetches that should be higher priority.

I think it is probably a better default to put it in hover/focus.

Reply:
IMO that's too late. Maybe we could add IntersectionObservers to load them when the link is visible, if you value network priority optimization over having less client JS running.

Reply:
Hmm, I don't really want to pull in an intersection observer here, and it probably wouldn't actually help with the nav case I'm thinking about (since the nav is generally visible on screen from the start).

We can try the fetch on mount. Worst case, if someone needs to avoid loading lazy entry points on mount all these functions are exposed as hooks so they can build a `Link` with different `fetchEntryPoint` behavior.

Reply:
I wrapped the prefetch with requestIdleCallback/requestAnimationFrame for now, but maybe we could expose some options to let the users decide.